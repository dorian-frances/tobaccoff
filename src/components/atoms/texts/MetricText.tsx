import { Text } from 'react-native-paper';
import { FontsEnum } from '../../../assets/fonts/fonts.enum';

export type MetricTextProps = {
  metric: number;
  unit: string;
  minimumFractionDigits: number;
  fontSize: number;
};

const MetricText = ({
  metric = 40.0,
  unit = 'â‚¬',
  minimumFractionDigits = 2,
  fontSize = 40,
}: MetricTextProps) => {
  return (
    <Text style={{ fontFamily: FontsEnum.BOLD, fontSize: fontSize }}>
      {metric.toLocaleString('fr-FR', {
        minimumFractionDigits: minimumFractionDigits,
      })}
      {unit}
    </Text>
  );
};

export default MetricText;
