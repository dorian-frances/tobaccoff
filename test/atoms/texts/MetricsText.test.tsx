import React from 'react';
import { render } from '@testing-library/react-native';
import MetricText from '../../../src/components/atoms/texts/MetricText';

describe('<MetricText/>', () => {
  const fakeMetric = 100;
  const fakeUnit = 'â‚¬';
  const fakeMinimumFractionDigit = 2;
  const fakeFontSize = 20;
  it('should render correctly', () => {
    const component = render(
      <MetricText
        metric={fakeMetric}
        unit={fakeUnit}
        minimumFractionDigits={fakeMinimumFractionDigit}
        fontSize={fakeFontSize}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
