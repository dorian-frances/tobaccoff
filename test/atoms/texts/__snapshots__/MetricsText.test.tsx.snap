// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetricText/> should render correctly 1`] = `
<Text
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "fontFamily": "System",
        "fontWeight": "400",
        "letterSpacing": 0,
      },
      {
        "writingDirection": "ltr",
      },
      {
        "fontFamily": "Montserrat-Bold",
        "fontSize": 20,
      },
    ]
  }
>
  100,00
  â‚¬
</Text>
`;
